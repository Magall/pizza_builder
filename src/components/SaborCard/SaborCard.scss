#saborCard{
    display: flex;
    justify-content: space-between;
    border:1px solid white;
    border-radius: 8px;
    padding:8px;
    color: #000;
    background:white;
    font-weight: bold;
    margin-bottom: $small-spacing;
    p{
        text-transform: capitalize;
    }
    button{
        // background: white;
       font-weight: bolder;
       color: black;
       font-size: 18px;
       border-radius: 16px;
       border: 1px solid white
  
      

    }
    
}