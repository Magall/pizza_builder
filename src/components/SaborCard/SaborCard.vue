<template>
  <div id="saborCard">
    <p>{{ sabor }}</p>
    <div class="sabores">
      <button class="first" @click="apagarSabor()" v-if="this.saborInserido()">
        -
      </button>
      <button id="plus" v-show="this.saboresCheio()" @click="inserirSabor()">
        +
      </button>
    </div>
  </div>
</template>
<style lang="scss" scoped>
@import "./SaborCard.scss";
</style>
<script>
import store from "@/store";
export default {
  props: ["sabor"],
  computed: {
    sabores: {
      get() {
        return store.getters.getSaboresEscolhidos.length;
      },
    },
  },
  methods: {
    saboresCheio() {
      if (this.sabores < 2) {
        return true;
      } else {
        return false;
      }
    },
    inserirSabor() {
      store.commit("adicionarSabor", this.sabor);
    },
    apagarSabor() {
      store.commit("removerSabor", this.sabor);
    },
    saborInserido() {
      const sabores = store.getters.getSaboresEscolhidos;
      const index = sabores.indexOf(this.sabor);
      if (index === -1) {
        return false;
      } else {
        return true;
      }
    },
  },
};
</script>
