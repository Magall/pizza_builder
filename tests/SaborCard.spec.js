import { mount,createLocalVue } from "@vue/test-utils";
import SaborCard from "@/components/SaborCard/SaborCard";
import Vuex from 'vuex'
import "@testing-library/jest-dom";


describe("SaborCard", () => {
  test("Se clicar no adicionar - deve apararecer", async () => {
    const wrapper = mount(SaborCard);
    await wrapper.find("#plus").trigger("click");
    expect(wrapper.find(".first").element).toBeVisible();
  });

  test("Se adicionar dois sabores botÃ£o adicionar some ", async() => {
 
    const wrapper = mount(SaborCard);
    await wrapper.find("#plus").trigger("click");
    await wrapper.find("#plus").trigger("click");
    expect(wrapper.find("#plus").element).not.toBeVisible();
  });
});
