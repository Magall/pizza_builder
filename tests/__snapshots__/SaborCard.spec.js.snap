// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaborCard Se dois sabores foram selecionados + some 1`] = `
"<div id=\\"saborCard\\">
  <p></p>
  <div class=\\"sabores\\"><button class=\\"first\\">-</button> <button id=\\"plus\\">+</button></div>
</div>"
`;
